File: app/src/main/java/com/nextcloud/talk/models/json/capabilities/UserStatusCapability.java
package com.nextcloud.talk.models.json.capabilities;

import com.bluelinelabs.logansquare.annotation.JsonField;
import com.bluelinelabs.logansquare.annotation.JsonObject;

import org.parceler.Parcel;

@Parcel
@JsonObject
public class UserStatusCapability {
    @JsonField(name = "enabled")
    boolean enabled;

    @JsonField(name = "supports_emoji")
    boolean supportsEmoji;

    public boolean isEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    public boolean isSupportsEmoji() {
        return supportsEmoji;
    }

    public void setSupportsEmoji(boolean supportsEmoji) {
        this.supportsEmoji = supportsEmoji;
    }
}

File: app/src/main/java/com/nextcloud/talk/models/json/capabilities/UserStatusCapability.kt
/*
 * Nextcloud Talk - Android Client
 *
 * SPDX-FileCopyrightText: 2022 Tim Krüger <t@timkrueger.me>
 * SPDX-FileCopyrightText: 2017-2019 Mario Danic <mario@lovelyhq.com>
 * SPDX-License-Identifier: GPL-3.0-or-later
 */
package com.nextcloud.talk.models.json.capabilities

import android.os.Parcelable
import com.bluelinelabs.logansquare.annotation.JsonField
import com.bluelinelabs.logansquare.annotation.JsonObject
import kotlinx.parcelize.Parcelize
import kotlinx.serialization.Serializable

@Parcelize
@JsonObject
@Serializable
data class UserStatusCapability(
    @JsonField(name = ["enabled"])
    var enabled: Boolean,
    @JsonField(name = ["restore"])
    var restore: Boolean,
    @JsonField(name = ["supports_emoji"])
    var supportsEmoji: Boolean
) : Parcelable {
    // This constructor is added to work with the 'com.bluelinelabs.logansquare.annotation.JsonObject'
    constructor() : this(false, false, false)
}
